### DISCUSSION TITLE
For a multi-objective study, parameter `n_trials` is silently disabled?

### DISCUSSION THREAD

**Discussion by pangjac** [optuna/optuna#5090] [Q&A] [answered]
### Don't use GitHub Issues to ask support questions.

I have customized a objective function customized_metrics where it returns multiple objective to tune.
I set the `n_trials` to **1000**, hoping that I can have a **1000 iterations of trials**. I specifically set timeout to 1000 to avoid time out and no configurations of prune is enabled.

After the optuna finished, `len(study.trials)` only returns **26**, and `len(study.best_trials)` is **7**.
I can understawnd that `len(study.best_trials)` > 1, since I applied a multiple objective to tune so for certain trials, it might be better at loss but the other better at acc.

But what about `len(study.trials)` ? **I expected `len(study.trials)` should be 1000, which should be equal to `n_trials`, no**?

P.S If this question is not appropriate in **Issues**, but **Discussions**, please let me know. I am new to our community so I might not be familiar enough with our general discussion rules. 

Thank you!

```
# define multiple objective
def customized_metrics(y_true, y_pred):
    # some logic
    return loss, accuracy, mean_abs_error

# dtrain and dvalid should be separately init out side of objective function
def objective_internal(trial, dtrain, dvalid)
    #set params 
    # Train the model
    output = xgb.dask.train(client, params=param, dtrain=dtrain, num_boost_round=100, obj=CusL1_objective)
    # The trained model
    bst = output['booster']
    preds = xgb.dask.predict(client, bst, dvalid)
    
    loss, acc, mae = customized_metrics(y_test_dd.to_dask_array(lengths=True), preds)
    return  loss, acc, mae

from functools import partial
def objective_wrapper(dtrain, dvalid):
    return partial(objective_internal, dtrain = dtrain, dvalid=dvalid)

objective = objective_wrapper(dtrain, dvalid)

from dask.distributed import wait
study = optuna.create_study(directions=["minimize", "maximize", "minimize"])
study.optimize(objective, n_trials=1000, timeout=1000)
```

**Comment by not522** [comment, marked as answer]
I guess that this is because you set `timeout`. The document says "The study continues to create trials until the number of trials reaches `n_trials`, `timeout` period elapses, `stop()` is called, or a termination signal such as SIGTERM or Ctrl+C is received". https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html#optuna.study.Study.optimize
